pypeit/coadd2d.py

```diff
@@ -885,6 +886,17 @@ def compute_offsets(self, offsets):
 
         """
         msgs.info('Get Offsets')
+        # 0) offsets are provided in the header of the spec2d files
+        if offsets== 'header':
+            msgs.info('Using offsets from header')
+            pscale = self.stack_dict['detectors'][0].platescale
+            dithoffs = [self.spectrograph.get_meta_value(f, 'dithoff') for f in self.spec2d]
+            if None in dithoffs:
+                msgs.error('Dither offsets keyword not found for one or more spec2d files. '
+                           'Choose another option for `offsets`')
+            dithoffs_pix = - np.array(dithoffs) / pscale
+            self.offsets = dithoffs_pix[0] - dithoffs_pix
+            self.offsets_report(self.offsets, 'header keyword')
```

Comment:
I think it would be cleaner logic if this was 

if offset == 'header'

elif self.objid_bri is None ad offsets = 'auto'

elif isinstance(offsets, (list, np.ndarray))

else: 
msgs.error("Invalid value for offsets"

Or something like that. These are mutually exclusive.

Reply:
Also shouldn't masked offsets be here. And shouldn't the offsets report be printed out for all?

Reply:
Thanks @jhennawi . I followed your suggestion and I rearranged the if statements.